tests/astimezone_test.py

```diff
@@ -0,0 +1,247 @@
+        ),
+    ]
+)
+def test_astimezone(
+    original_dt: datetime_tz, target_tz: ZoneInfo, expected_dt: datetime_tz
+) -> None:
+    """Test that astimezone returns the correct datetime with proper timezone conversion."""
+    result = original_dt.astimezone(target_tz)
+
+    # Check that the result is of the same class as the original
+    assert isinstance(result, datetime_tz)
+
+    # Check that the timezone conversion is correct
+    assert result.year == expected_dt.year
+    assert result.month == expected_dt.month
+    assert result.day == expected_dt.day
+    assert result.hour == expected_dt.hour
+    assert result.minute == expected_dt.minute
+    assert result.second == expected_dt.second
+    assert result.microsecond == expected_dt.microsecond
+    assert result.tzinfo == expected_dt.tzinfo
+
+
+def test_astimezone_converts_type() -> None:
+    """Test that astimezone converts to datetime_tz when changing timezones."""
+    utc_dt = datetime_utc(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('UTC'))
+    result_utc = utc_dt.astimezone(ZoneInfo('CET'))
+    assert isinstance(result_utc, datetime_tz)
+
+    cet_dt = datetime_cet(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('CET'))
+    result_cet = cet_dt.astimezone(ZoneInfo('UTC'))
+    assert isinstance(result_cet, datetime_tz)
+
+    local_dt = datetime_local.now()
+    result_local = local_dt.astimezone(ZoneInfo('CET'))
+    assert isinstance(result_local, datetime_tz)
+
+    tz_dt = datetime_tz(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('UTC'))
+    result_tz = tz_dt.astimezone(ZoneInfo('CET'))
+    assert isinstance(result_tz, datetime_tz)
+
+
+def test_deepcopy_astimezone() -> None:
+    """Test that astimezone works correctly with deepcopy."""
+    original = datetime_utc(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('UTC'))
+
+    # Deep copy the original
+    copied = deepcopy(original)
+
+    # Apply astimezone to the copy
+    result = copied.astimezone(ZoneInfo('CET'))
+
+    # Check that the result is correct
+    assert isinstance(result, datetime_tz)
+    assert result.hour == 11  # UTC+1 for CET in January
+    assert result.tzinfo == ZoneInfo('CET')
+
+
+def test_astimezone_edge_cases() -> None:
+    """Test astimezone with edge cases like DST transitions."""
+    # Test around DST transition (CET to CEST)
+    # March 28, 2021 - clocks go forward at 2:00 AM
+    before_dst = datetime_utc(2021, 3, 28, 1, 30, 0, tzinfo=ZoneInfo('UTC'))
+    after_dst = datetime_utc(2021, 3, 28, 2, 30, 0, tzinfo=ZoneInfo('UTC'))
+
+    # Convert to CET
+    before_dst_cet = before_dst.astimezone(ZoneInfo('CET'))
+    after_dst_cet = after_dst.astimezone(ZoneInfo('CET'))
+
+    assert isinstance(before_dst_cet, datetime_tz)
+    assert isinstance(after_dst_cet, datetime_tz)
+
+    # The hour difference should be 1 for before DST and 2 for after DST
+    assert before_dst_cet.hour == 3  # 1:30 UTC + 1 hour = 3:30 CET
+    assert after_dst_cet.hour == 4   # 2:30 UTC + 2 hours = 4:30 CEST
```

Comment:
I don't think this test really tests anything. Also, these comments seem wrong:

```
    # The hour difference should be 1 for before DST and 2 for after DST
    assert before_dst_cet.hour == 3  # 1:30 UTC + 1 hour = 3:30 CET
    assert after_dst_cet.hour == 4   # 2:30 UTC + 2 hours = 4:30 CEST
```

1:30 UTC + 1 hour is not 3:30 CET
And why would this logic suddently make the zoneinfo CEST for the second datetime? Even though Central Europe switches to CEST on daylight savings, you can still express a time in CET, which I believe is what is happening here anyways.